modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayRoutinePermissions.testcase

```diff
@@ -0,0 +1,223 @@
+			JSONTestrayAPI.deleteProject(projectName = ${projectName});
+
+			Testray.tearDownCP();
+
+			ClientExtensionGeneral.goToRemoteAppsPortlet();
+
+			ClientExtensionGeneral.deleteRemoteApp(tableEntry = "Testray");
+		}
+
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Story ID: LPS-178269 | Test Summary: Verify that the User with the 'Testray Analyst' role should not be able to create/edit/delete the routine but can view"
+	@priority = 3
+	test AssertAnalystRoleCanNotCreateEditAndDeleteRoutine {
+		property test.name.skip.portal.instance = "TestrayRoutinePermissions#AssertAnalystRoleCanNotCreateEditAndDeleteRoutine";
+
+		task ("Given that the user login with the Testray Analyst role") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "analyst@testray.com",
+				userLoginFullName = "Analyst User");
+
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
+
+		task ("And can see the routine in the table") {
+			AssertElementPresent(
+				key_projectName = ${routineName},
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("And can not see add button") {
+			AssertElementNotPresent(locator1 = "Button#PLUS");
+		}
+
+		task ("When right click in routine") {
+			RightClick.rightClick(
+				key_projectName = ${routineName},
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("Then can not see the 'Edit, Delete' options in right click") {
+			AssertElementNotPresent(
+				item = "Edit",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+
+			AssertElementNotPresent(
+				item = "Delete",
+				locator1 = "TestrayContextMenu#CONTEXT_MENU_PROJECTS_ITEMS");
+		}
+
+		task ("And can not see the kebab menu") {
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+
+			AssertElementNotPresent(locator1 = "ObjectAdmin#KEBAB_MENU");
+		}
+	}
+
+	@description = "Story ID: LPS-178269 | Test Summary: Verify that the User with the 'Testray Lead' role should not be able to create/edit/delete the routine but can view"
+	@priority = 3
+	test AssertLeadRoleCanNotCreateEditAndDeleteRoutine {
+		property test.name.skip.portal.instance = "TestrayRoutinePermissions#AssertLeadRoleCanNotCreateEditAndDeleteRoutine";
+
+		task ("Given that the user login with the Testray Lead role") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "lead@testray.com",
+				userLoginFullName = "Lead User");
+
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the routine page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+		}
```

Comment:
This block can be placed next to the Given block. To give you more context about the use of Given, usually in this block you can add all the steps up to the test intention page.

Reply:
Done